src/ssh/writer.rs

```diff
@@ -173,8 +173,8 @@ impl Writer {
                 self.write_bytes(&key.key);
             },
             PublicKeyKind::Rsa(key) => {
-                self.write_bytes(&key.n);
-                self.write_bytes(&key.e);
+                self.write_mpint(&key.e);
```

Comment:
Still investigating this.

Reply:
I believe this is correct! Good catch!

Reply:
I'm still looking into it but it seems you're right as the reader also reads `e` before `n`. Tearing apart an SSH certificate, it also appears it does that as well.

Did you find this because you tried to use SSH certs and it didn't work? If so, sorry!